File: lua/raindrops/raindrops.lua
local raindrops = {}

function raindrops(n)
    result = ""
    if n % 3 == 0 then result = result .. 'Pling' end
    if n % 5 == 0 then result = result .. 'Plang' end
    if n % 7 == 0 then result = result .. 'Plong' end
    if result == "" then 
        return tostring(n) 
    else
        return result
    end
end

return raindrops

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    fun convert(n: Int): String {
        val pling = if (n % 3 == 0) "Pling" else ""
        val plang = if (n % 5 == 0) "Plang" else ""
        val plong = if (n % 7 == 0) "Plong" else ""
        val all = "$pling$plang$plong"
        return all.ifEmpty { "$n" }
    }
}
